File: Array/liuchang/day19/day19.java
public class day19 {

    static class MyQueue{
        char data[] = new char[20];
        int front = 0;
        int rear = 0;

        public void pushQueue(MyQueue myQueue,String s){
            myQueue.data = s.toCharArray();
            myQueue.rear = data.length;
        }

        public void removeDuplicates(MyQueue myQueue){
            int num = -1;
            for (int k = 0; k < myQueue.rear-1; ++k) {
                if (myQueue.data[k] == myQueue.data[k + 1]) {
                    num = k;
                    break;
                }
            }
            if (num != -1) {
                char temp;
                int i, j;
                for (i = num; i < num + 2; ++i) {
                    temp = myQueue.data[num];
                    for (j = num; j < myQueue.rear-1; ++j) {
                        myQueue.data[j] = myQueue.data[j + 1];
                    }
                    myQueue.data[j] = temp;
                }
                myQueue.rear = myQueue.rear - 2;
                removeDuplicates(myQueue);
            }
        }
    }

    public static void main(String[] args) {
        MyQueue myQueue = new MyQueue();
        myQueue.pushQueue(myQueue,"abbaca");
        for (int i = 0; i < myQueue.rear ; i++) {
            System.out.println("====="+myQueue.data[i]);
        }
        myQueue.removeDuplicates(myQueue);
        for (int i = 0; i < myQueue.rear ; i++) {
            System.out.println("<===>"+myQueue.data[i]);
        }
    }

}

File: Array/liuchang/day19/day19.kt
package net.study.kotlintest

/**
 * @author       刘畅
 * @createdate   2020/6/22
 * @describe
 */
class day19 {
    class MyQueue {
        var data = CharArray(20);
        var front = 0;
        var rear = 0;

        fun pushQueue(myQueue: MyQueue, data: String) {
            myQueue.data = data.toCharArray();
            myQueue.rear = myQueue.data.size;
        }

        fun removeDuplicates(myQueue: MyQueue) {
            var num = -1;
            for (k in 0 until myQueue.rear) {
//            for (int k = 0; k < myQueue.rear-1; ++k) {
                if (myQueue.data[k] == myQueue.data[k + 1] && k + 1 != myQueue.rear) {
                    num = k;
                    break;
                }
            }
            if (num != -1) {
                var temp: Char;
//                for (i = num; i < num + 2; ++i) {
                for (i in num until num + 2) {
                    temp = myQueue.data[num];
                    for (j in num until myQueue.rear - 1) {
//                    for (j = num; j < myQueue.rear - 1; ++j) {
                        myQueue.data[j] = myQueue.data[j + 1];
                    }
                    myQueue.data[myQueue.rear - 1] = temp;
                }
                myQueue.rear = myQueue.rear - 2;
                removeDuplicates(myQueue);
            }
        }


    }

    fun main(arg: Array<String>) {
        var myQueue: MyQueue = MyQueue();
        myQueue.pushQueue(myQueue, "abbaca");
        for (i in 0 until myQueue.rear) {
            System.out.println("=====" + myQueue.data[i]);
        }
        myQueue.removeDuplicates(myQueue);
        for (i in 0 until myQueue.rear) {
            System.out.println("<===>" + myQueue.data[i]);
        }
    }
}